core/java/iesi-core/src/test/java/io/metadew/iesi/script/action/http/HttpExecuteRequestTest.java

```diff
@@ -0,0 +1,453 @@
+
+        HttpComponent httpComponent = createBaseComponent(new ArrayList<>(), new ArrayList<>());
+        mockGetAndTraceHttpComponent(httpComponent);
+
+        HttpExecuteRequest httpExecuteRequest = new HttpExecuteRequest(executionControl, scriptExecution, actionExecution);
+
+        assertThrows(KeyValuePairException.class, httpExecuteRequest::prepare);
+    }
+
+    @Test
+    void prepareDefaultNoHeadersAndQuery() throws HttpRequestBuilderException, URISyntaxException {
+
+        ActionParameter requestActionParameter = createActionParameter("request", "request");
+        ActionParameter queryParameterActionParameter = createActionParameter("queryParameters", "name=name");
+        Action action = createAction(requestActionParameter, queryParameterActionParameter);
+        HttpQueryParameter defaultQueryParam = new HttpQueryParameter("status", "sold");
+
+        when(actionExecution.getAction())
+                .thenReturn(action);
+
+        mockResolvedValues(requestActionParameter, queryParameterActionParameter);
+
+        HttpComponent httpComponent = createBaseComponent(new ArrayList<>(), Collections.singletonList
+                (defaultQueryParam));
+        mockGetAndTraceHttpComponent(httpComponent);
+
+        HttpExecuteRequest httpExecuteRequest = new HttpExecuteRequest(executionControl, scriptExecution, actionExecution);
+        httpExecuteRequest.prepare();
+
+        HttpRequestBase httpRequest = httpExecuteRequest.getHttpRequest().getHttpRequest();
+
+        assertThat(httpRequest.getURI()).hasToString("https://hostendpoint?name=name&status=sold");
+    }
+
+
+    @Test
+    void prepareOverrideNoHeaderAndQuery() throws HttpRequestBuilderException, URISyntaxException {
+
+        ActionParameter requestActionParameter = createActionParameter("request", "request");
+        ActionParameter queryParametersActionParameter = createActionParameter("queryParameters", "status=sold");
+        Action action = createAction(requestActionParameter, queryParametersActionParameter);
+        HttpQueryParameter defaultQueryParameters = new HttpQueryParameter("status", "available");
+
+        when(actionExecution.getAction())
+                .thenReturn(action);
+
+        mockResolvedValues(requestActionParameter, queryParametersActionParameter);
+
+        HttpComponent httpComponent = createBaseComponent(new ArrayList<>(), Collections.singletonList(defaultQueryParameters));
+        mockGetAndTraceHttpComponent(httpComponent);
+
+        HttpExecuteRequest httpExecuteRequest = new HttpExecuteRequest(executionControl, scriptExecution, actionExecution);
+        httpExecuteRequest.prepare();
+
+        HttpRequestBase httpRequest = httpExecuteRequest.getHttpRequest().getHttpRequest();
+
+        assertThat(httpRequest.getURI()).hasToString("https://hostendpoint?status=sold");
+    }
+
+
+    @ParameterizedTest
+    @ValueSource(strings = {"status=sold", "status=sold,name=name"})
```

Comment:
why a parameterized unit test? it seems that for parameter 1 you already have the use case above, only parameter 2 (multiple headers) seems to be a applicable here